src/arrayMethodJoin.js

```diff
@@ -5,7 +5,34 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let separatorValue = separator;
+
+    if (separatorValue === undefined) {
+      separatorValue = ',';
+    }
+
+    const arr = this;
+    let result = '';
+
+    if (arr.length === 0) {
+      return '';
+    }
+
+    for (const element of arr) {
+      const str = element === null
+      || element === undefined ? '' : element.toString();
+
+      result += str;
+
+      if (arr.indexOf(element) < arr.length - 1) {
```

Comment:
If you go with the previously suggested changes this here could be shortened to:
```js
if (i < this.length - 1) {
  result += String(separator)
}
```